resources/asciidoctor/spec/open_in_widget_spec.rb

```diff
@@ -0,0 +1,167 @@
+# frozen_string_literal: true
+
+require 'open_in_widget/extension'
+
+RSpec.describe OpenInWidget do
+  before(:each) do
+    Asciidoctor::Extensions.register do
+      treeprocessor OpenInWidget
+    end
+  end
+
+  after(:each) do
+    Asciidoctor::Extensions.unregister_all
+  end
+
+  spec_dir = File.dirname(__FILE__)
+
+  def stub_file_opts(result)
+    return {
+      'copy_snippet' => proc { |uri, source| result << [uri, source] },
+      'write_snippet' => proc { |uri, snippet| result << [uri, snippet] },
+    }
+  end
+
+  %w[console sense kibana].each do |lang|
```

Comment:
I *is* ok that this doesn't have `autosense` because this widget only support `sense`. `autosense` is transformed into `sense` in tree compatibility processor.